src/main/java/core/interfaces/ITunableParameters.java

```diff
@@ -140,6 +140,6 @@ default void setParameterValues(Map<?, Object> values) {
      * @param json      The raw JSON detailing the parameters
      * @return          The instantiated object
      */
-    ITunableParameters registerChild(String name, JSONObject json);
+    Object registerChild(String name, JSONObject json);
```

Comment:
Why is this an Object now? Seems very general and I'm afraid of things breaking it.

Reply:
The javadoc is a bit out of date, so I'll correct this. (and it's taken me a while to remember why I did this - so definitely needs documenting)

The underlying reason is that registerChild() is used whenever we are extracting settings for an ITunableParameters from JSON. It is called when we find a JSONObject as a property in the JSON file. This change is for point 9) in the changes, specifically:

"Not everything in a JSON parameters file now needs to subclass TunableParameters. If a thing is instantiated from JSON that is not such a sub-class, then we stop further recursion to find tunable parameters, but still instantiate the thing. The best current example is heuristic or opponentHeuristic in MCTS. These can provide a specific class that implements IStateHeuristic without also needing to themselves be TunableParameters. (See next point for where this can be useful.)"